lib/xapi-stdext-date/date.ml

```diff
@@ -44,12 +44,21 @@ type iso8601 = Ptime.t
   match x |> Ptime.of_rfc3339 |> Ptime.rfc3339_error_to_msg with
   | Error (`Msg e) -> invalid_arg (Printf.sprintf "date.ml:of_string: %s" e)
   | Ok (t, tz, _)  -> match tz with
-                      | None | Some 0 -> t
+                      | None | Some 0 -> (t, None)
                       | Some _        -> invalid_arg (Printf.sprintf "date.ml:of_string: %s" x)
 
-let to_string t =
-  Ptime.to_rfc3339 ~tz_offset_s:0 (* to ensure Z printed, rather than +00:00 *) t |>
-  Astring.String.filter (fun char -> char <> '-') (* remove dashes for backwards compatibility *)
+let to_string (t, offset) =
+  let compat chars =
+      Astring.String.filter (fun c -> List.for_all (fun c' -> c <> c') chars)
+  in
+  let t_string =
+    (* to ensure Z printed, rather than +00:00 *)
+    Ptime.to_rfc3339 ~tz_offset_s:0 t
+  in
+  match offset with
+  (* in all cases, remove '-' for backwards compatibility. for localtime we must also remove 'Z' *)
+  | None   -> compat ['-'] t_string
+  | Some _ -> compat ['-'; 'Z'] t_string
```

Comment:
```
let t = Ptime_clock.now();;
val t : Ptime.t = <abstr>
let offset = Ptime_clock.current_tz_offset_s ();;
val offset : int option = Some 3600
Ptime.to_rfc3339 ~tz_offset_s:Option.(value ~default:0 offset) t;;
- : string = "2020-07-30T15:34:27+01:00"
Ptime.to_rfc3339 ~tz_offset_s:Option.(value ~default:0 None) t;;
- : string = "2020-07-30T14:34:27Z"
```

I think you can use `Ptime.to_rfc3339 ~tz_offset_s:Option.(value ~default:0 offset)` and then drop everything after a certain amount of chars (I think you expect the output string to be fixed in size), or alternatively after the last + or - or Z.

Reply:
That'll work for both local and UTC timestamps.

Reply:
I'm not sure I see the advantage in doing this? we don't care about the value of the offset, only whether there is one or not 

(offset = None <=> UTC , offset = Some <=> localtime, in fact I'll change the type of `iso8601` to make it more obvious )

Reply:
I was going to say that we don't need the actual time zone in the type, since we only care about UTC or local. Just a boolean would do. I saw that you just made an update along those lines already 👍